Sources/FluentPostgresDriver/FluentPostgresDatabase.swift

```diff
@@ -5,64 +5,43 @@ import PostgresKit
 import PostgresNIO
 import SQLKit
 
-struct _FluentPostgresDatabase {
-    let database: PostgresDatabase
+struct _FluentPostgresDatabase<E: PostgresJSONEncoder, D: PostgresJSONDecoder> {
+    let database: any SQLDatabase
     let context: DatabaseContext
-    let encoder: PostgresDataEncoder
-    let decoder: PostgresDataDecoder
+    let encodingContext: PostgresEncodingContext<E>
+    let decodingContext: PostgresDecodingContext<D>
     let inTransaction: Bool
-    let sqlLogLevel: Logger.Level
 }
 
 extension _FluentPostgresDatabase: Database {
     func execute(
         query: DatabaseQuery,
-        onOutput: @escaping (DatabaseOutput) -> ()
+        onOutput: @escaping (any DatabaseOutput) -> ()
     ) -> EventLoopFuture<Void> {
-        var expression = SQLQueryConverter(delegate: PostgresConverterDelegate())
-            .convert(query)
-        switch query.action {
-        case .create where query.customIDKey != .string(""):
-            expression = PostgresReturningID(
-                base: expression,
-                idKey: query.customIDKey ?? .id
-            )
-        default: break
-        }
-        let (sql, binds) = self.serialize(expression)
-        self.logger.log(level: self.sqlLogLevel, "\(sql) \(binds)")
```

Comment:
For reference for anyone else reviewing - this is now handled [in PostgresKit](https://github.com/vapor/postgres-kit/blob/main/Sources/PostgresKit/PostgresDatabase%2BSQL.swift#L43-L45)